src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -272,5 +297,49 @@ public async Task CheckoutRemoteRefAsync(string commit)
             await RunGitAsync("fetch", "origin", commit);
             await RunGitAsync("checkout", commit);
         }
+
+        internal IImmutableList<AssetData> GetAssetData(string asset1Name, string asset1Version, string asset2Name, string asset2Version)
+        {
+            ImmutableList<AssetData> sourceAssets = ImmutableList<AssetData>.Empty;
+            AssetData asset1 = new AssetData(false)
+            {
+                Name = asset1Name,
+                Version = asset1Version,
+                Locations = ImmutableList<AssetLocationData>.Empty
+            };
+            asset1.Locations.Add(new AssetLocationData(LocationType.NugetFeed)
```

Comment:
Locations is Immutable.  Calling "Add" on an Immutable list doesn't mutate it.  You should probably just assign it in one go with `ImmutableList.Create(new AssetLocationData...`

And at that point you don't have to keep recreating the sourceAsset list too, the whole method could be
```csharp
return ImmutableList.Create(
    new AssetData(false) 
    {
        // ....
        Locations = ImmutableList.Create(new AssetLocationData() //...
    },
    new AssetData(false) 
    {
        // ....
        Locations = ImmutableList.Create(new AssetLocationData() //...
    });
```

Reply:
Huh. That makes a lot more sense than what the MSDN page said to do, I like it. :)
https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablelist-1?view=netcore-3.1

Reply:
Yeah, all the ImmutableXxx<T> all have ImmutableXxx.Create<T> and ImmutableXxx.CreateBuilder<T> type things. Those are way more useful than the Add style methods, since those let you add bunch of stuff and then sort of "lock' the Xxx when you are done with the adding.